test/Driver/single_stack_test.jl

```diff
@@ -0,0 +1,615 @@
+# - this method is only called at `t=0`
+# - `state.ρcT`, `state.ρ`, ... are available here because we've 
+# specified `ρcT` in `vars_state_conservative`
+function init_state_conservative!(
+    m::SingleStack,
+    state::Vars,
+    aux::Vars,
+    coords,
+    t::Real,
+)
+    z = aux.z
+    ε1 = rand(Normal(0, m.σ))
+    ε2 = rand(Normal(0, m.σ))
+    state.ρ = 1
+    ρu = 1 - 4 * (z - m.zmax / 2)^2 + ε1
+    ρv = 1 - 4 * (z - m.zmax / 2)^2 + ε2
+    ρw = 0
+    state.ρu = SVector(ρu, ρv, ρw)
+
+    state.ρcT = state.ρ * m.c * aux.T
+end;
+
+# The remaining methods, defined in this section, are called at every
+# time-step in the solver by the [`BalanceLaw`](@ref
+# ClimateMachine.DGMethods.BalanceLaw) framework.
+
+# Overload `update_auxiliary_state!` to call `heat_eq_nodal_update_aux!`, or
+# any other auxiliary methods
+function update_auxiliary_state!(
+    dg::DGModel,
+    m::SingleStack,
+    Q::MPIStateArray,
+    t::Real,
+    elems::UnitRange,
+)
+    nodal_update_auxiliary_state!(heat_eq_nodal_update_aux!, dg, m, Q, t, elems)
+    return true # TODO: remove return true
+end;
+
+# Compute/update all auxiliary variables at each node. Note that
+# - `aux.T` is available here because we've specified `T` in
+# `vars_state_auxiliary`
+function heat_eq_nodal_update_aux!(
+    m::SingleStack,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+)
+    aux.T = state.ρcT / (state.ρ * m.c)
+end;
+
+# Since we have second-order fluxes, we must tell `ClimateMachine` to compute
+# the gradient of `ρcT` and `u`. Here, we specify how `ρcT`, `u` are computed. Note that
+#  - `transform.ρcT` and `transform.u` are available here because we've specified `ρcT` 
+# and `u`in `vars_state_gradient`
+function compute_gradient_argument!(
+    m::SingleStack,
+    transform::Vars,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+)
+    transform.ρcT = state.ρcT
+    transform.u = state.ρu / state.ρ
+end;
+
+# Specify where in `diffusive::Vars` to store the computed gradient from
+# `compute_gradient_argument!`. Note that:
+#  - e.g., `diffusive.α∇ρcT` is available here because we've specified `α∇ρcT` in
+#  `vars_state_gradient_flux`
+#  - e.g., `∇transform.ρcT` is available here because we've specified `ρcT`  in
+#  `vars_state_gradient`
+#  - `diffusive.μ∇u` is built using an anisotropic diffusivity tensor
+function compute_gradient_flux!(
+    m::SingleStack,
+    diffusive::Vars,
+    ∇transform::Grad,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+)
+    diffusive.α∇ρcT = m.α * ∇transform.ρcT
+    diffusive.μ∇u = Diagonal(@SVector [m.μh, m.μh, m.μv]) * ∇transform.u
+end;
+
+# Introduce Rayleigh friction towards a target profile as a source.
+# Note that:
+# - Rayleigh damping is only applied in the horizontal by subtracting
+#  the vertical component.
+function source!(
+    m::SingleStack,
+    source::Vars,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    args...,
+)
+    ẑ = [0.0, 0.0, 1.0]
```

Comment:
```suggestion
    ẑ = @SVector FT[0, 0, 1]
```
If you want this to be able to run on GPU